pkg/runtime/interfaces.go

```diff
@@ -30,6 +30,11 @@ const (
 	APIVersionInternal = "__internal"
 )
 
+type GroupVersioner interface {
+	VersionForGroup(group string) (unversioned.GroupVersion, bool)
```

Comment:
I don't think this interface is rich enough for what you need it for.  The decision about `Group` and `GroupVersion` are both tied to the originating `GroupVersionKind`.  For instance:

`v1.batch` has `Job`.  `v2alpha1.batch` has `Job` and `ScheduledJob`.  The decision of version for a given Kind depends on the kind you're talking about.  `ScheduledJob.__internal.batch` should be `ScheduleJob.v2alpha1.batch`, but `Job.__internal.batch` should be `Job.v1.batch`.